pkg/scalers/azure/azure_app_insights_test.go

```diff
@@ -0,0 +1,176 @@
+package azure
+
+import (
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+)
+
+type testExtractAzAppInsightsTestData struct {
+	testName      string
+	isError       bool
+	expectedValue int32
+	info          AppInsightsInfo
+	metricResult  ApplicationInsightsMetric
+}
+
+func mockAppInsightsInfo(aggregationType string) AppInsightsInfo {
+	return AppInsightsInfo{
+		MetricID:        "testns/test",
+		AggregationType: aggregationType,
+	}
+}
+
+func mockAppInsightsMetric(metricName, aggregationType string, value *float64) ApplicationInsightsMetric {
+	metric := ApplicationInsightsMetric{
+		Value: map[string]interface{}{
+			metricName: map[string]interface{}{},
+		},
+	}
+
+	if value == nil {
+		metric.Value[metricName].(map[string]interface{})[aggregationType] = nil
+	} else {
+		metric.Value[metricName].(map[string]interface{})[aggregationType] = *value
+	}
+
+	return metric
+}
+
+func newMetricValue(f float64) *float64 {
+	return &f
+}
+
+var testExtractAzAppInsightsData = []testExtractAzAppInsightsTestData{
+	{"metric not found", true, -1, mockAppInsightsInfo("avg"), mockAppInsightsMetric("test/test", "avg", newMetricValue(0.0))},
+	{"metric is nil", true, -1, mockAppInsightsInfo("avg"), mockAppInsightsMetric("testns/test", "avg", nil)},
+	{"incorrect aggregation type", true, -1, mockAppInsightsInfo("avg"), mockAppInsightsMetric("testns/test", "max", newMetricValue(0.0))},
+	{"success round down value", false, 5, mockAppInsightsInfo("max"), mockAppInsightsMetric("testns/test", "max", newMetricValue(5.2))},
+	{"success round up value", false, 5, mockAppInsightsInfo("max"), mockAppInsightsMetric("testns/test", "max", newMetricValue(5.5))},
```

Comment:
Round down 5.2 is 5 but round up 5.5 should be 6. Could this be a typo?

Reply:
That was actually a bug in the code that extracts the value. Thanks for catching that.